src/main/java/com/iota/iri/service/milestone/impl/MilestoneServiceImpl.java

```diff
@@ -95,6 +111,18 @@ public void updateMilestoneIndexOfMilestoneTransactions(Hash milestoneHash, int
         updateMilestoneIndexOfMilestoneTransactions(milestoneHash, index, index, new HashSet<>());
     }
 
+    /**
+     * {@inheritDoc}
+     * <br />
+     * We redirect the call to {@link #resetCorruptedMilestone(int, Set)} while initiating the set of {@code
+     * processedTransactions} with an empty {@link HashSet} which will ensure that we reset all found
+     * transactions.<br />
+     */
+    @Override
+    public void resetCorruptedMilestone(int index) throws MilestoneException {
+        resetCorruptedMilestone(index, new HashSet<>());
+    }
+
     @Override
     public MilestoneValidity validateMilestone(TransactionViewModel transactionViewModel, SpongeFactory.Mode mode,
```

Comment:
we should consider pulling out the `MilestoneVM.store()` from the validator -- a validator should be stateless.

Reply:
That's old legacy code that just got copied to the new class - refactoring everything was not part of the scope of the PR. The whole method can and should be refactored (also in relation to the cyclomatic complexity).

Reply:
sure - that was more of a comment, will open an issue.